pkg/collector/corechecks/servicediscovery/module/impl_linux_test.go

```diff
@@ -525,7 +525,7 @@ func assertCPU(t *testing.T, url string, pid int) {
 	// Calling getServicesMap a second time us the CPU usage percentage since the last call, which should be close to gopsutil value.
 	portMap := getServicesMap(t, url)
 	assert.Contains(t, portMap, pid)
-	assert.InDelta(t, referenceValue, portMap[pid].CPUCores, 0.10)
+	assert.InDelta(t, referenceValue, portMap[pid].CPUCores*100, 0.10)
```

Comment:
You should divide the referenceValue or fix the InDelta value, otherwise the InDelta value is not big enough.

Also, a comment here in the code (similar to the one in your PR description) about the difference in the semantics of the two values would be useful.